### DISCUSSION TITLE
Render custom component with FlexRender and Svelte 5

### DISCUSSION THREAD

**Discussion by moalamri** [TanStack/table#5713] [Q&A] [answered]
Hi,
Previously it was possible to render a custom cell using flexRender function, but now all I  can see in TanStack Table, FlexRender is a component that requires both content and context which are provided by `cell.column.columnDef.cell` and `
cell.getContext()`.

When I try to use a component in those two params it fails with an error about component anchor.
Even if I use cell prop as a returned component it also fails.

How to achieve that?

**Comment by KevinVandy** [comment]
Are you following the v9 examples on GitHub?

**Reply by moalamri** [reply]
Hi @KevinVandy 
Thank you for answering. I do follow the example on alpha branch, I don't see any example of rendering a custom cell component.

**Reply by moalamri** [reply, marked as answer]
I found it, thank you so much.
```javascript
import { renderComponent } from "@tanstack/svelte-table";
import TextCell from './TextCell.svelte'

const column = {
    accessorKey: 'id',
    cell: ({ cell }) => renderComponent(TextCell, { cell })
}
```
Inside TextCell
```svelte
<script lang="ts">
	import type { Cell } from '@tanstack/svelte-table';

	const { cell }: { cell: Cell<any, any> } = $props();
</script>

<div class:text-slate-300={!cell.getValue()}>
	{cell.getValue() || '--'}
</div>
```

I would appreciate if you add an example for that as it took me a while to figure it out :)

**Reply by c-kirkeby** [reply]
Have you found a way to transform the value before it gets passed as a cell? For example if we need to do `cell: (info) => new Date(info.getValue() as string).toLocaleString()` it gets a runtime error because it's not a `Cell` type any more.

**Reply by moalamri** [reply]
`cell` is a property of `info` you should  destruct it or user it as following:

`Date(info.cell.getValue()).toLocaleString()`
